Remove hints page (#9627)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md

## Quick hits

> - dbt now surfaces occasional, non-blocking CLI hints that suggest ways to optimize your project. Hints are enabled by default. Disable them with `--no-hints-enabled` or `hints_enabled: false` in `dbt_project.yml`.

Reviewer: ```suggestion
- dbt now surfaces occasional, non-blocking hints that suggest ways to optimize your project. Hints are enabled by default. Disable them with `hints_enabled` global config.
```

Reviewer: would be nice to then link to global config page
